pkg/tfbridge/names_test.go

```diff
@@ -248,6 +248,34 @@ func TestBijectiveNameConversion(t *testing.T) {
 				"remaped":     "plural_types",
 			},
 		},
+		{ // Check when SchemaInfo is non-nil
+			schema: map[string]*schemav2.Schema{
+				"gallery_application":  {Type: schemav2.TypeList},
+				"gallery_applications": {Type: schemav2.TypeList},
+			},
+			info: map[string]*SchemaInfo{
+				"gallery_application":  {},
+				"gallery_applications": {},
+			},
+			expected: map[string]string{
+				"galleryApplication":  "gallery_application",
+				"galleryApplications": "gallery_applications",
+			},
+		},
+		{ // Check when SchemaInfo.MaxItemsOne: &false is set explicitly
+			schema: map[string]*schemav2.Schema{
+				"gallery_application":  {Type: schemav2.TypeList},
+				"gallery_applications": {Type: schemav2.TypeList},
+			},
+			info: map[string]*SchemaInfo{
+				"gallery_application":  {MaxItemsOne: ref(false)},
```

Comment:
Does this do anything without schemav2.Schema also specifying MaxItems=1? or merely adding MIO=false was tripping something up?

Reply:
Yeah got it